tests/conftest.py

```diff
@@ -63,6 +64,43 @@ def _run_cli_command(command: click.Command, options: list = None, raises: bool
     return _run_cli_command
 
 
+@pytest.fixture
+def generate_input_generator_cls():
+    """Return a factory to create a subclass of an ``InputGenerator``."""
+
+    def _generate_input_generator_cls(inputs_dict=None):
+        """Generate a subclass of ``InputGenerator``.
+
+        :param inputs_dict: an optional dictionary of inputs to be defined on the process spec.
+        :param namespaces: an optional list of namespaces to be defined on the process spec.
+        """
+        from aiida_common_workflows.generators import InputGenerator
+
+        class TestInputGenerator(InputGenerator):
+            """Test subclass of ``InputGenerator``."""
+
+            _protocols = {'moderate': {'description': 'bla'}}
+            _default_protocol = 'moderate'
+
+            @classmethod
+            def define(cls, spec):
+                super().define(spec)
+
+                if inputs_dict is not None:
+                    for k, val in inputs_dict.items():
+                        spec.input(k, valid_type=val)
+
+            def _construct_builder(self, **kwargs) -> engine.ProcessBuilder:
+                if 'mutable' in kwargs:
+                    kwargs['mutable']['test'] = 'whatever'
```

Comment:
would remove this. It is specific to one test but really difficult to figure out which one. Should be locally implemented. See other comment